File: src/main/java/de/schramm/royalbash/tdd/core/card/NoOpCard.java
package de.schramm.royalbash.tdd.core.card;

import de.schramm.royalbash.tdd.core.Card;
import de.schramm.royalbash.tdd.core.Context;
import de.schramm.royalbash.tdd.core.Game;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class NoOpCard implements Card {

    private final String name;

    @Override
    public Game invoke(Context context) {
        return context.getGame();
    }
}

File: src/main/kotlin/de/schramm/royalbash/domain/card/spell/NoOpCard.kt
package de.schramm.royalbash.domain.card.spell

import de.schramm.royalbash.domain.Card
import de.schramm.royalbash.domain.Context
import de.schramm.royalbash.domain.Game

class NoOpCard (
        override val id: String,
        override val name: String = "NoOp Card",
        override val text: String = "NoOp Card Text",
        override val cost: Int = 0
) : Card {

    override fun invoke(context: Context): Game {
        return context.game
    }
}
